errors.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package mutex
+
+import (
+	"github.com/juju/errors"
+)
+
+var (
+	ErrTimeout   = errors.New("timeout acquiring mutex")
```

Comment:
Acquiring a mutex can fail for other reasons. Yes we don't expect them to fail for those reasons, but they are valid none the less. The caller can at least check that the cause of the error was the Timeout they expected, or something more fatal. That was my reasoning